src/game/server/vscript_server.cpp

```diff
@@ -1201,6 +1200,57 @@ static void Script_Say( HSCRIPT hPlayer, const char *pText )
 	}
 }
 
+//This is a vscript function that converts RGB into a transmittable format for colored text
+static char* Script_TextColor(int R, int G, int B)
+{
+	//Force channel ranges between 0 - 255
+	if (R > 255)
+	{
+		R = 255;
+	}
+	else if (R < 0)
+	{
+		R = 0;
+	}
+
+	if (G > 255)
+	{
+		G = 255;
+	}
+	else if (G < 0)
+	{
+		G = 0;
+	}
+
+	if (B > 255)
+	{
+		B = 255;
+	}
+	else if (B < 0)
+	{
+		B = 0;
+	}
+	
+	//create float modifiers at range 0 - 255 for conversion output
+	float outputMod_R = R / 255.0f;
+	float outputMod_G = G / 255.0f;
+	float outputMod_B = B / 255.0f;
+
+	//This char must match the char used for the client's COLOR_INPUTCUSTOMCOL - in my case it's 0x08
+	char Con_Char = (char)0x08; 
+	//pass float modifiers multiplied by max ASCII translation base then increment by 32 which is float ASCII offset
+	char R_Char = (char)(32 + (outputMod_R * 94));
+	char G_Char = (char)(32 + (outputMod_G * 94));
+	char B_Char = (char)(32 + (outputMod_B * 94));
+
+
+	//Format a string with results as 4 characters.
+	char outputChars[5];
+	Q_snprintf(outputChars, sizeof(outputChars), "%c%c%c%c", Con_Char, R_Char, G_Char, B_Char);
+
+	return outputChars;
```

Comment:
Return of a pointer to the out-of-scope variable outputChars.

Reply:
Are you suggesting something like the following?

	char outputChars[5];
	Q_snprintf(outputChars, sizeof(outputChars), "%c%c%c%c", Con_Char, R_Char, G_Char, B_Char);

	return ScriptVariant_t(outputChars, true);